os/mm/mm_heap/mm_heapinfo_parse_heap.c

```diff
@@ -105,6 +203,8 @@ void heapinfo_parse_heap(FAR struct mm_heap_s *heap, int mode, pid_t pid)
 	FAR struct mm_freenode_s *fnode;
 #endif
 
+	heapinfo_check_corruption(heap);
```

Comment:
In this case, we walk the whole heap twice. How about merging this with the line from 219 to 286?
Or how about exiting the parse heap when there is a corruption? Because if there is corrupted node, it is not useful to see the whole heap.

Reply:
I want to keep the check corruption api separate. Also, even if there is a heap corruption, it is possible that we can continue to execute until the corrupted node actually causes a problem. So, I think its ok to print the corrupted node info and then proceed to print the heapinfo as requested by the user. If there is a problem in some node, and heapinfo does not work properly, then it is ok because we have already printed the corrupted node info.